driver/maindriver.ml

```diff
@@ -44,6 +44,8 @@ let main argv ppf =
         exit 2
       end
     end;
+    if Clflags.(should_stop_after Compiler_pass.Lambda)
+      then raise (Compenv.Exit_with_status 0);
```

Comment:
Reading the code of the exception handler at the end of this file, my impression is that this shortcut-with-an-exception approach interacts badly with the `-dprofile` option: my guess would be that `-dprofile` does not work correctly if `stop-after lambda` is used.

I think that you should declare a new exception in this module, for example `Continue`, to stop early, instead of reusing `Exit_with_status` to also signal correct termination.